### PR TITLE
Theory manual and some doco clean up

### PR DESCRIPTION
**Pull request by ragusa**
The theory manual has been ported from latex. The process is 95% automatic but the figures had to be fixed by hand. 

Some additional cleanup to the doco was made while adding the theory manual

### CODE REVIEW

**Comment by andrsd** [comment]
Add this patch, so that the github action can generate the doco:

```
diff --git a/doc/requirements.txt b/doc/requirements.txt
index 48ae5445..94971b73 100644
--- a/doc/requirements.txt
+++ b/doc/requirements.txt
@@ -3,3 +3,4 @@ myst-parser
 sphinx
 sphinx-design
 sphinx-rtd-theme
+sphinxcontrib-bibtex
```

File: doc/theory/parallelization.rst [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+The content below is only intended to be a very brief introduction to
+the basic concepts.
+
+Aggregation
+-----------
+
+In the previous definitions of the multigroup iterative algorithms, the
+description of the solution techniques were given on a group-by-group
+basis. Likewise, the solution of a single-group problem problem was
+given direction-by -direction. At the level of a given group :math:`g`,
```
Comment (andrsd) [inline doc/theory/parallelization.rst:33]: Extra space in "direction-by -direction"

File: doc/theory/parallelization.rst [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+for a given direction :math:`d`, a full- domain transport sweep is
```
Comment (andrsd) [inline doc/theory/parallelization.rst:34]: Extra space in "full- domain"

**Review by andrsd — CHANGES_REQUESTED** [review]

File: doc/theory/discretization.rst [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+     %\sum_{\ell=0}^{L_{\text{max}}} \sum_{m=-\ell}^{m=\ell} 
+     \sum_{\ell,m} \, \frac{2\ell+1}{4\pi}\sigma_{s,\ell}(\vec{r}) Y_{\ell,m}(\vec{\Omega}) \phi_{\ell,m}(\vec{r})
+     +
+     Q(\vec{r},\vec{\Omega})  \,,
+
+  where the source term :math:`Q(\vec{r},\vec{\Omega})` is now the sum
+  of the external source and the upscattering and downscattering
+  contributions to the current group :math:`g`.
+
+Angle Discretization (the :math:`S_n` method)
```
[the remark this answers is not in the document]
Comment (ragusa) [reply doc/theory/discretization.rst:83]: I have simplified the title to just have Sn in plain text then.

File: doc/CMakeLists.txt [hunk 4]
```diff
@@ -52,7 +52,16 @@ if(DOXYGEN_FOUND AND NOT ${SPHINX_BUILD} STREQUAL SPHINX_BUILD-NOTFOUND)
...
+    configure_file(getting_started.rst src/getting_started.rst)
+    configure_file(theory/index.rst src/theory/index.rst)
+    configure_file(theory/background.rst src/theory/background.rst)
+    configure_file(theory/discretization.rst src/theory/discretization.rst)
+    configure_file(theory/cross_sections.rst src/theory/cross_sections.rst)
+    configure_file(theory/outcome.rst src/theory/outcome.rst)
+    configure_file(theory/iterative.rst src/theory/iterative.rst)
+    configure_file(theory/parallelization.rst src/theory/parallelization.rst)
+    configure_file(theory/adjoint.rst src/theory/adjoint.rst)
+    configure_file(${CMAKE_SOURCE_DIR}/tutorials/index.rst src/tutorials/index.rst)
```
[the remark this answers is not in the document]
Comment (ragusa) [reply doc/CMakeLists.txt:64]: I actually add forgotten to add that new index file that is located in the `tutorials` folder. good catch

**Comment by ragusa** [comment]
comments resolved. requirements.txt updated. thanks

**Review by andrsd — APPROVED** [review]

**Comment by ragusa** [comment]
@zhardy-lanl @wdarylhawkins : reviewing doco is not necessarily easy as it's a bunch of rst text. But you can always checkout my branch, go to the `build/` folder, configure, and then `make doc`. If, in the past, you built the doco locally, you likely have most of the python requirements satisfied but you will need to pip install the latest addition to the requirements, sphinxcontrib-bibtex.

**Review by zhardy-lanl — APPROVED** [review]

**Review by wdhawkins — APPROVED** [review]

**Pull request merged**